src/ripple/nodestore/ShardPool.md

```diff
@@ -0,0 +1,68 @@
+# Shard Manager
+
+## Overview
+
+Shard NuDB and SQLite databases consume server resources. This can be unnecessarily taxing on servers with many shards. The shard manager class aims to reduce this by intelligently managing open databases. The manager will maintain a limited pool of open databases prioritized by use. The following sections describe the shard manager class in greater detail.
+
+### ShardManager
+
+The ShardManager class is privately implemented in the nodestore namespace. Its public interface is thread safe and provides functions to add and remove shards along with information about their state. The manager maintains a map of all server shard instances and their states. This map is comprised of ShardInfo structs keyed on shard indexes. To keep track of open databases, a small map is used as a pool of open databases prioritized on last use time stamps. Upon the necessity of a new connection exceeding the pool open limit, the shard with the oldest time stamp will be closed and removed to make room for the new connection.
+
+```C++
+public:
+	enum class ShardState
+	{
+	    none,
+	    final,      // Immutable, complete and validated
+	    acquire,    // Being acquired
+	    import,     // Being imported
+	    finalize    // Being finalized
+	};
+
+	void
+	add(std::unique_ptr<Shard>&&);
+
+	void
+	remove(std::uint32_t);
+
+	void
+	removeAll();
+
+ 	std::shared_ptr<Shard>
+	get(std::uint32_t);
+
+	bool
+	setState(std::uint32_t);
+
+	optional<ShardState>
+	getState(std::uint32_t);
+
+    void
+    for_each(std::function <void(std::shared_ptr<Shard> const&)>);
+
+private:
+	struct ShardInfo
```

Comment:
The `ShardInfo` struct was removed right?

Reply:
Yes, I will update the document.